src/Pages/form/Components/ArrivalOnset.tsx

```diff
@@ -63,55 +68,70 @@ const OnsetStyledToggleButtonGroup = withStyles(() => ({
   },
 }))(ToggleButtonGroup);
 
-const ArrivalOnset: React.FC = () => {
+interface PatientProps {
+  arrivalDate: Date | null;
+  arrivalTime: Date | null;
+  clearDate: Date | null;
+  clearTime: Date | null;
+  lastDate: Date | null;
+  lastTime: Date | null;
+  firstDate: Date | null;
+  firstTime: Date | null;
+  onset: string;
+}
+
+interface Props {
+  values: PatientProps;
+  fieldName: string;
+  onChange: (field: string, value: any, shouldValidate?: boolean) => void;
+}
+
+const ArrivalOnset = (props: Props) => {
   const classes = useStyle();
+  const { values, fieldName, onChange } = props;
 
-  //* Arrival
-  const [arrivalDate, setArrivalDate] = React.useState<Date | null>(null);
-  //* Onset
-  const [onset, setOnset] = React.useState<string | null>("");
-  //+ clear onset
   const [showClearPicker, setShowPickerClear] = useState(false);
-  const [clearDate, setClearDate] = React.useState<Date | null>(null);
-  //+ unknown onset
   const [showUnknownPicker, setShowPickerUnknown] = useState(false);
-  //- last seen
-  const [lastDate, setLastDate] = React.useState<Date | null>(null);
-  //- first seen
-  const [firstDate, setFirstDate] = React.useState<Date | null>(null);
-
-  //* Onset
-  // toggle button
+  const handlePickerChange = (name: string) => (value: Date) => {
+    if (name === "arrivalDate") {
+      values.clearDate = null;
```

Comment:
I think it does not work because you try to direct set value to formik value right?. It is the problem same with previous. Please create a temporary variable and copy formik values. You can change directly to that variable.